File: Ser2025/Company/Company.Web/Modules/Company/StateProvince/StateProvinceColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.StateProvince")]
[BasedOnRow(typeof(StateProvinceRow), CheckNames = true)]
public class StateProvinceColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int StateProvinceId { get; set; }
    [EditLink]
    public string StateProvinceCode { get; set; }
    public string CountryCodeCountryName { get; set; }
    public string StateProvinceName { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/StateProvinceColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StateProvinceRow } from "./StateProvinceRow";

export interface StateProvinceColumns {
    StateProvinceId: Column<StateProvinceRow>;
    StateProvinceCode: Column<StateProvinceRow>;
    CountryCodeCountryName: Column<StateProvinceRow>;
    StateProvinceName: Column<StateProvinceRow>;
}

export class StateProvinceColumns extends ColumnsBase<StateProvinceRow> {
    static readonly columnsKey = 'Company.StateProvince';
    static readonly Fields = fieldsProxy<StateProvinceColumns>();
}